src/main/java/org/opensearch/security/transport/SecurityInterceptor.java

```diff
@@ -162,7 +163,9 @@ public <T extends TransportResponse> void sendRequestDecorate(
 
         final boolean isDebugEnabled = log.isDebugEnabled();
 
-        final boolean isSameNodeRequest = localNode != null && localNode.equals(connection.getNode());
+        final boolean isSameNodeRequest = localNode != null
+            && localNode.equals(connection.getNode())
+            && (options.type() == null || !options.type().equals(TransportRequestOptions.Type.STREAM));
```

Comment:
Why is this change needed? Wouldn't behavior for sameNodeRequest be the same for any transport? i.e. retain the existing threadcontext headers and avoid ser/de?

Reply:
we do not have direct channel support (to avoid ser/de) yet in stream transport, once we have it, then we can enable it here.

Reply:
I would recommend to add a comment explaining this, or possibly slightly refactoring this. This is a bit confusing, as the intialized variable is called `isSameNodeRequest` - and the fact that this request comes from the same node does not change. Just the channel changes.

Reply:
Can you separate the clauses out please:
```java
final boolean isSameNodeRequest = localNode != null
            && localNode.equals(connection.getNode());
final boolean shouldSkipIfNotStreamChannel =  options.type() == null || !options.type().equals(TransportRequestOptions.Type.STREAM));     
 
```            
and then use these?

Reply:
done